browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,263 @@
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ false);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+gfx::Size SplitViewLocationBar::CalculatePreferredSize(
+    const views::SizeBounds& available_size) const {
+  auto size = views::WidgetDelegateView::CalculatePreferredSize(available_size);
+  if (auto* view = view_observation_.GetSource(); view && view->width()) {
+    size.set_width(std::min(view->width(), size.width()));
+  }
+  return size;
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::DidChangeVisibleSecurityState() {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    auto url = contents->GetVisibleURL();
+    auto origin = url::Origin::Create(url);
+    auto formatted_origin = url_formatter::FormatUrl(
+        origin.opaque() ? url : origin.GetURL(),
+        url_formatter::kFormatUrlOmitDefaults |
+            url_formatter::kFormatUrlOmitHTTPS,
+        base::UnescapeRule::SPACES, nullptr, nullptr, nullptr);
+    url_->SetText(formatted_origin);
+    UpdateIcon();
+  } else {
+    url_->SetText({});
+    UpdateIcon();
+  }
+
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::UpdateIcon() {
+  // At the moment, we show only warning icon.
+  safety_icon_->SetVisible(!IsContentsSafe());
+}
+
+bool SplitViewLocationBar::IsContentsSafe() const {
+  if (!web_contents()) {
+    return true;
+  }
+
+  if (SecurityStateTabHelper::FromWebContents(web_contents())
+          ->GetSecurityLevel() == security_state::SecurityLevel::SECURE) {
+    return true;
+  }
+
+  auto url = web_contents()->GetVisibleURL();
```

Comment:
seems safer to use [GetLastCommittedURL](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_contents.h;bpv=1;bpt=1;l=430?q=getvisibleurl&ss=chromium&gsn=GetLastCommittedURL&gs=KYTHE%3A%2F%2FKswCCqMBa3l0aGU6Ly9jaHJvbWl1bS5nb29nbGVzb3VyY2UuY29tL2NvZGVzZWFyY2gvY2hyb21pdW0vc3JjLy9tYWluP2xhbmc9YyUyQiUyQj9wYXRoPWNvbnRlbnQvcHVibGljL2Jyb3dzZXIvd2ViX2NvbnRlbnRzLmgjVllEVHd1UWY5QmY5SW8zdk1NWVViYzExRU0xYVY0TU5zNlVZOFBoc0FUOAqjAWt5dGhlOi8vY2hyb21pdW0uZ29vZ2xlc291cmNlLmNvbS9jb2Rlc2VhcmNoL2Nocm9taXVtL3NyYy8vbWFpbj9sYW5nPWMlMkIlMkI_cGF0aD1jb250ZW50L3B1YmxpYy9icm93c2VyL3dlYl9jb250ZW50cy5oI2ZfVEhXLUQ2bm55aVdLaUd0TGIzVUwwazctLUU5Nmh0SUhRQTlEeTdQVUE%3D)

Reply:
I think we probably want `GetVisibleURL` here @diracdeltas - its what the `LocationBarModelDelegate` and the `CustomTabBar` are using (or equivalent).

The `CustomTabBar` goes directly through `GetVisibleURL` while the `ChromeLocationBarModelDelegate` does the following:

```cpp
content::NavigationEntry* ChromeLocationBarModelDelegate::GetNavigationEntry()
    const {
  content::NavigationController* controller = GetNavigationController();
  return controller ? controller->GetVisibleEntry() : nullptr;
}

bool ChromeLocationBarModelDelegate::GetURL(GURL* url) const {
  DCHECK(url);
  content::NavigationEntry* entry = GetNavigationEntry();
  if (!entry || entry->IsInitialEntry())
    return false;

  *url = entry->GetVirtualURL();
  return true;
}
```

for reference, this is the implementation of `GetVisibleURL` in `WebContentsImpl`

```cpp
const GURL& WebContentsImpl::GetVisibleURL() {
  // We may not have a navigation entry yet.
  NavigationEntry* entry = GetController().GetVisibleEntry();
  return entry ? entry->GetVirtualURL() : GURL::EmptyGURL();
}
```

and this is the comment on `GetVisibleURL` in `web_contents.h`

```cpp
  // Gets the virtual URL currently being displayed in the URL bar, if there is
  // one. This URL might be a pending navigation that hasn't committed yet, so
  // it is not guaranteed to match the current page in this WebContents.
  virtual const GURL& GetVisibleURL() = 0;
```

I think this makes sense - if you're on `https://mybank.com` and you somehow navigate to `https://evil-phising-mybank.com` you don't want to be showing `https://mybank.com` in the URL bar while the phising site is loading.

Reply:
@bridiver do you have thoughts here? 

my reasoning is that this mini/readonly URL display should reflect what page is currently loaded, whereas the full URL bar (because the user may enter URLs into it) has to show pending navigations.

that is if you're on google.com in a split view, and the user initiates a navigation to othersite.com which takes a long time to load, it makes sense to me that the urlbar would show othersite.com but i'd still expect the split view's readonly URL display to show google.com. however i could go either way on this.

note that there have been historically a bunch of URL spoofing issues like https://issues.chromium.org/issues/40096119 with the omnibox

Reply:
> my reasoning is that this mini/readonly URL display should reflect what page is currently loaded, whereas the full URL bar (because the user may enter URLs into it) has to show pending navigations.

I agree with this. Unlike the omnibox, the mini URL bar is for the inactive tab, so we should prioritize what's actually loaded over what the user typed or what's still loading(pending).

Reply:
okay, sgtm :smile: